pkg/controller/hvpa/hvpa_controller.go

```diff
@@ -631,9 +650,47 @@ func getVpaWeightFromIntervals(hvpa *autoscalingv1alpha1.Hvpa, desiredReplicas,
 	return vpaWeight
 }
 
-func (r *ReconcileHvpa) scaleIfRequired(hpaStatus *autoscaling.HorizontalPodAutoscalerStatus, vpaStatus *vpa_api.VerticalPodAutoscalerStatus, hvpa *autoscalingv1alpha1.Hvpa, deployment *appsv1.Deployment) (int32, bool, autoscalingv1alpha1.VpaWeight, error) {
+func (r *ReconcileHvpa) scaleIfRequired(hpaStatus *autoscaling.HorizontalPodAutoscalerStatus, vpaStatus *vpa_api.VerticalPodAutoscalerStatus, hvpa *autoscalingv1alpha1.Hvpa, target runtime.Object) (int32, bool, autoscalingv1alpha1.VpaWeight, error) {
+
+	var newObj runtime.Object
+	var deploy *appsv1.Deployment
+	var ss *appsv1.StatefulSet
+	var ds *appsv1.DaemonSet
+	var rs *appsv1.ReplicaSet
+	var rc *corev1.ReplicationController
+	var currentReplicas int32
+	var podSpec *corev1.PodSpec
+
+	kind := target.GetObjectKind().GroupVersionKind().Kind
+	targetCopy := target.DeepCopyObject()
+
+	switch kind {
+	case "Deployment":
+		deploy = targetCopy.(*appsv1.Deployment).DeepCopy()
+		currentReplicas = *deploy.Spec.Replicas
+		podSpec = &deploy.Spec.Template.Spec
+	case "Statefulset":
+		ss = targetCopy.(*appsv1.StatefulSet).DeepCopy()
+		currentReplicas = *ss.Spec.Replicas
+		podSpec = &ss.Spec.Template.Spec
+	case "DaemonSet":
+		ds = targetCopy.(*appsv1.DaemonSet).DeepCopy()
+		currentReplicas = *deploy.Spec.Replicas
+		podSpec = &deploy.Spec.Template.Spec
+	case "ReplicaSet":
+		rs = targetCopy.(*appsv1.ReplicaSet).DeepCopy()
```

Comment:
```suggestion
		rs = targetCopy.(*appsv1.ReplicaSet)
```
The `DeepCopyObject` [above](https://github.com/gardener/hvpa-controller/pull/6/files#diff-bfa21ee442000583bc7aa6f53ec3660cR665) should be sufficient.